bundle/config/mutator/load_git_details.go

```diff
@@ -21,50 +20,41 @@ func (m *loadGitDetails) Name() string {
 }
 
 func (m *loadGitDetails) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-	// Load relevant git repository
-	repo, err := git.NewRepository(b.BundleRoot)
+	info, err := git.FetchRepositoryInfo(ctx, b.BundleRoot, b.WorkspaceClient())
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	// Read branch name of current checkout
-	branch, err := repo.CurrentBranch()
-	if err == nil {
-		b.Config.Bundle.Git.ActualBranch = branch
-		if b.Config.Bundle.Git.Branch == "" {
-			// Only load branch if there's no user defined value
-			b.Config.Bundle.Git.Inferred = true
-			b.Config.Bundle.Git.Branch = branch
-		}
-	} else {
-		log.Warnf(ctx, "failed to load current branch: %s", err)
+	b.WorktreeRoot = info.WorktreeRoot
+
+	b.Config.Bundle.Git.ActualBranch = info.CurrentBranch
```

Comment:
We perform validation that `branch == actual_branch` in `ValidateGitDetails`. Can we completely get rid of `ValidateGitDetails` and `ActualBranch` and just perform the validation here?

I see `ValidateGitDetails` is only called during deploy but I think it's valid to perform this validation in all bundle commands and not just `databricks bundle deploy`.

Reply:
Also please use a diagnostic and add the location / path information to the error if we are already touching this piece of the code

Reply:
Thanks for the suggestions. In this PR I don’t want to touch on the logic of brach/actual_branch and similar variables. There is a separate PR to tweak that behavior that I’ll update once this is merged (https://github.com/databricks/cli/pull/1940).

I also think that ValidateGitDetails can be merged in here but again, it’s out of scope here, will be addressed in follow up PR.